app/celery/process_delivery_status_result_tasks.py

```diff
@@ -265,3 +276,247 @@ def sms_status_update(
+
+    Parameters:
+        status (str): Current notification status.
+        retries (int): The number of retries already attempted.
+        max_retries (int): The maximum number of retries allowed.
+        sent_at (datetime): The timestamp when the message was initially sent.
+        retry_window (timedelta): The allowable time window for retries.
+
+    Returns:
+        bool: True if retrying is allowed, False otherwise.
+    """
+    # Calculate the time elapsed since the message was sent
+    time_elapsed = datetime.datetime.utcnow() - sent_at
+
+    return (
+        (retries is not None)
+        and (status == NOTIFICATION_SENDING)
+        and (retries <= max_retries)
+        and (time_elapsed < retry_window)
+    )
+
+
+def get_sms_retry_delay(retry_count: int) -> int:
+    """Calculate the retry delay for SMS delivery with random jitter.
+
+    Delays were chosen to cover transient errors and small scale service disruptions
+
+    Note: Delays limited to under 900s due to AWS SQS limitation
+
+    Assumes up to 2 retries, repeating final delay for subsequent:
+    1st retry: 60 seconds +/- 10%
+    2nd retry: 10 minutes (600 seconds) +/- 10%
+
+    Parameters:
+        retry_count (int): The retry attempt number.
+
+    Returns:
+        int: Delay in seconds with applied jitter.
+    """
+    delay_with_jitter_by_retry_count = {
+        1: (60, 6),  # 60 seconds +/- 6 seconds (10%)
+        2: (600, 60),  # 10 minutes +/- 60 seconds (10%)
+    }
+
+    # Safeguard against retry counts outside the defined range
+    # Default to largest delay
+    default_delay_with_jitter = (600, 60)
+
+    base_delay, jitter_range = delay_with_jitter_by_retry_count.get(retry_count, default_delay_with_jitter)
+
+    # Apply jitter
+    delay = int(base_delay + random.randint(-jitter_range, jitter_range))  # nosec non-cryptographic use case
+
+    return delay
+
+
+def update_sms_retry_count(
+    notification_retry_id: str,
+    initial_value: int = 0,
+    ttl: int | None = None,
+) -> int:
+    """Get updated retry count for this notification from redis store, initializing pre-increment initial value if it doesn't exist
+
+    Args:
+        notification_retry_id (str): The value to retrieve from redis.
+        initial_value (int): The initial value to set if the key doesn't exist.
+        ttl (int, optional): Time-to-live for the value, in seconds.
+
+    Returns:
+        int: The current retry count
+
+    Raises:
+        ValueError: Unable to retrieve from redis
+    """
+
+    try:
+        # Set the initial value only if the value does not exist
+        redis_store.set(notification_retry_id, initial_value, ex=ttl, nx=True)
+        value = redis_store.incr(notification_retry_id)
+        return int(value)
+    except (ValueError, TypeError):
+        raise ValueError(
+            f"Expected an integer value for id '{notification_retry_id}', but got: {value} (type: {type(value)})"
+        )
+    except Exception:
```

Comment:
Please use the [redis base exception](https://redis-py.readthedocs.io/en/stable/exceptions.html) if you are not going to be specific.

Reply:
Missed this.